rapids-cmake/cpm/spdlog.cmake

```diff
@@ -62,6 +62,10 @@ function(rapids_cpm_spdlog)
   include("${rapids-cmake-dir}/cpm/detail/generate_patch_command.cmake")
   rapids_cpm_generate_patch_command(spdlog ${version} patch_command)
 
+  # Add fmt in the case where we need to fetch spdlog from source
+  include("${rapids-cmake-dir}/cpm/fmt.cmake")
+  rapids_cpm_fmt(BUILD_EXPORT_SET spdlog INSTALL_EXPORT_SET spdlog)
```

Comment:
I'm not sure if this is the proper way to be doing this, but I figure it would have the following behavior:
- In the case where we're finding spdlog from a conda package, this will find fmt as well
- In the case where we're finding spdlog not from a conda package, this will find fmt if it is already in the system, otherwise it will fetch it and build it (as far as I can tell there isn't a CMake option to avoid building a library for fmt)
- In the case where we're fetching spdlog, this will find fmt if it is already in the system, otherwise it will fetch it and build it

This generally seems like the correct behavior that we want.

I'm not sure if setting the exports as I've done here for `spdlog` is correct, but without these tests would fail from what looked like trying to export spdlog targets.

Reply:
The problem with this approach is that it won't obey `spdlog_ROOT` and the associated search behavior. If a project has extenerally built spdlog and fmt. Using `spdlog_ROOT` would cause any internal find calls in `spdlog-config.cmake` to first search beside it before looking globally.

I think we need to do the following:

1. Have the current call to `rapids_cpm_fmt` be opt-in.
2. Do `list(APPEND fmt_ROOT ${spdlog_ROOT})` before calling `rapids_cpm_fmt`
3. Set `SPDLOG_FMT_EXTERNAL_HO` only when opting in

Reply:
> The problem with this approach is that it won't obey `spdlog_ROOT` and the associated search behavior. If a project has extenerally built spdlog and fmt. Using `spdlog_ROOT` would cause any internal find calls in `spdlog-config.cmake` to first search beside it before looking globally.

Are you saying that if someone sets `spdlog_ROOT` that it should search for the `fmt` dependency of spdlog beside `spdlog_ROOT` before using the standard search paths? That feels a bit funky to me, but if that's standard CMake behavior so be it.

Will tackle those 3 bullets.

Reply:
> Are you saying that if someone sets spdlog_ROOT that it should search for the fmt dependency of spdlog beside spdlog_ROOT before using the standard search paths?

Yes when the `find_package` call for fmt is inside the `spd` config.cmake file. This is the purpose of the `<PackageName>_ROOT` variables they are meant as a way to re-root the search behavior so that all dependency package queries first search 'beside' the primary project.

Reply:
Should be addressed in the latest commits, PTAL

Reply:
Added some comments, but overall looks good